tests/phpunit/tests/style-engine/wpStyleEngineCssDeclarations.php

```diff
@@ -0,0 +1,235 @@
+			),
+			$css_declarations->get_declarations()
+		);
+	}
+
+	/**
+	 * Should compile css declarations into a css declarations block string.
+	 */
+	public function test_generate_css_declarations_string() {
+		$input_declarations = array(
+			'color'                  => 'red',
+			'border-top-left-radius' => '99px',
+			'text-decoration'        => 'underline',
+		);
+		$css_declarations   = new WP_Style_Engine_CSS_Declarations( $input_declarations );
+
+		$this->assertSame(
+			'color:red;border-top-left-radius:99px;text-decoration:underline;',
+			$css_declarations->get_declarations_string()
+		);
+	}
+
+	/**
+	 * Should compile css declarations into a prettified css declarations block string.
+	 */
+	public function test_generate_prettified_css_declarations_string() {
+		$input_declarations = array(
+			'color'                  => 'red',
+			'border-top-left-radius' => '99px',
+			'text-decoration'        => 'underline',
+		);
+		$css_declarations   = new WP_Style_Engine_CSS_Declarations( $input_declarations );
+
+		$this->assertSame(
+			'color: red; border-top-left-radius: 99px; text-decoration: underline;',
+			$css_declarations->get_declarations_string( true )
+		);
+	}
+
+	/**
+	 * Should compile css declarations into a prettified and indented css declarations block string.
+	 */
+	public function test_generate_prettified_with_indent_css_declarations_string() {
+		$input_declarations = array(
+			'color'                  => 'red',
+			'border-top-left-radius' => '99px',
+			'text-decoration'        => 'underline',
+		);
+		$css_declarations   = new WP_Style_Engine_CSS_Declarations( $input_declarations );
+
+		$this->assertSame(
+			'	color: red;
+	border-top-left-radius: 99px;
+	text-decoration: underline;',
+			$css_declarations->get_declarations_string( true, 1 )
+		);
+	}
+
+	/**
+	 * Should compile css declarations into a css declarations block string.
+	 */
+	public function test_generate_prettified_with_more_indents_css_declarations_string() {
+		$input_declarations = array(
+			'color'                  => 'red',
+			'border-top-left-radius' => '99px',
+			'text-decoration'        => 'underline',
+		);
+		$css_declarations   = new WP_Style_Engine_CSS_Declarations( $input_declarations );
+
+		$this->assertSame(
+			'		color: red;
+		border-top-left-radius: 99px;
+		text-decoration: underline;',
+			$css_declarations->get_declarations_string( true, 2 )
+		);
+	}
+
+	/**
+	 * Should escape values and run the CSS through safecss_filter_attr.
+	 */
+	public function test_remove_unsafe_properties_and_values() {
+		$input_declarations = array(
+			'color'        => 'url("https://wordpress.org")',
+			'font-size'    => '<red/>',
+			'margin-right' => '10em',
+			'padding'      => '</style>',
+			'potato'       => 'uppercase',
+		);
+		$css_declarations   = new WP_Style_Engine_CSS_Declarations( $input_declarations );
+
+		$this->assertSame(
+			'margin-right:10em;',
+			$css_declarations->get_declarations_string()
+		);
```

Comment:
```suggestion
		$mock_action = new MockAction();
		add_filter( 'safe_style_css', array( $mock_action, 'filter' );
		
		$this->assertSame(
			1,
			$mock_action->get_call_count(),
			'"safe_style_css" filters were not applied.'
		);
		
		$this->assertSame(
			'margin-right:10em;',
			$css_declarations->get_declarations_string(),
			'Unsafe properties and values were not removed'
		);
```
Part of this tests that the returned value has been filtered correctly. While the current implementation uses `safecss_filter_attr()`, a future change could find another approach to achieve the same result.

As `safecss_filter_attr()` applies `safe_style_css` filters, such a change would break BC if `safe_style_css` filters may not be applied in a new implementation.

To protect BC, this should also test that `safe_style_css` filters have been run, as shown above.

Additionally, the current test should be evaluated to check if `safecss_filter_attr_allow_css` filters currently run. If so, this should also be covered.